.github/workflows/release.yaml

```
@@ -18,11 +23,23 @@ jobs:
       - uses: actions/setup-python@v5
       - name: Install dependencies
         run: |
-          python -m pip install -U pip wheel setuptools setuptools-scm twine      
-        # python -m pip install -U pip wheel "setuptools<66.0.0" setuptools-scm twine
+          python -m pip install -U build check-manifest pip pytest setuptools setuptools-scm twine wheel
       - name: Build distributions
-        run: python setup.py sdist bdist_wheel
+        run: python -m build --sdist --wheel . --outdir dist
+
+      - name: CheckFiles
+        run: |
+          ls dist
+          check-manifest --verbose
+
+      - name: Test wheels
+        run: >
+          cd dist
+          && python -m twine check *
+          && python -m pip install *.whl
+          && cp --recursive ../tests .
+          && python -m pytest -rxs tests
```

I don't think you need the `&&` if you don't fold everything into one line:
```suggestion
        run: |
          cd dist
          python -m twine check *
          python -m pip install *.whl
          cp --recursive ../tests .
          pytest -rxs tests
```

---

I believe that, if something errors out, it continues with the next command in your form.

---

The [default shell](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_iddefaultsrunshell) is `bash -e`, which will fail the whole thing on the first command that fails.

And as a test:
```
$ bash -e <<EOF
false
echo yes
EOF
```
prints nothing as the first command failed.